### ISSUE TITLE
When vaadin grid is hidden, and then I resized the window. Data from the grid are cleared.

### ISSUE THREAD

**Issue by nexuslifeline** [vaadin/vaadin-grid#978]

**Comment by nexuslifeline** [comment]
Based on what I observed, the _update method on the _gridResizeHandler causes this. I don't really understand what is does but when I've tried to comment this method, the problem seems to not occur.

**Comment by nathan-collins** [comment]
you can fix this by doing a `clearCache()`

**Comment by nexuslifeline** [comment]
I dont think that this is a cached issue. But will try what you say. Thank you.

**Comment by nexuslifeline** [comment]
By the way this is the actual scenario. 

1.) The initial page load which is basically at browser full screeen. Within the page, there are 2 instance of vaadin grid on the Active users list and deactivated users list.
![1](https://user-images.githubusercontent.com/16952835/30307402-8523b3f4-97b0-11e7-9460-b52c5304ed70.png)

2.) And then when I tried to resized the browser window. The first time I've resized the window, I can see the deactivated users list.
![2](https://user-images.githubusercontent.com/16952835/30307476-f77a1e70-97b0-11e7-8d51-e76c4d919bca.png)

3.) Here's the issue. When I go back to active users list Tab, and then I resized again the window. When I tried to go again to Deactivated users list tab, Vaadin grid is empty.
![3](https://user-images.githubusercontent.com/16952835/30307526-4eaa6b6e-97b1-11e7-9b80-bd184572fc92.png)

**Comment by moderndeveloperllc** [comment]
This may seem silly, but do all the grids have different element IDs? I've run into issues where duplicate IDs can cause issues with iron-pages and things hiding/showing. FWIW, I can go between iron-pages on my app resize and everything just fine, so I'm not thinking this is with the vaadin element (but who knows right?).

**Comment by stat92** [comment]
Grid deletes scroll listener if it's not visible on resize. You can find it here:
https://github.com/vaadin/vaadin-grid/blob/master/src/iron-list.html#L1058
If it was your problem, you can observe visibility and add scroll listener when grid will be visible again.
